src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriverDetailsVisitor.java

```diff
@@ -0,0 +1,55 @@
+package edu.kis.powp.jobs2d.drivers.visitor;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.commons.Pair;
+import edu.kis.powp.jobs2d.drivers.DriversComposite;
+import edu.kis.powp.jobs2d.drivers.LoggerDriver;
+import edu.kis.powp.jobs2d.drivers.RecordingDriverDecorator;
+import edu.kis.powp.jobs2d.drivers.UsageMonitorDriverDecorator;
+import edu.kis.powp.jobs2d.features.RecordFeature;
+
+public class DriverDetailsVisitor implements DriverVisitor {
+    private String details = "";
+
+    public String getDetails() { return this.details; }
+
+    public String getDetails(Job2dDriver driver) {
+        if(driver instanceof LoggerDriver) {
+            visit((LoggerDriver) driver);
+        } else if(driver instanceof DriversComposite) {
+            visit((DriversComposite) driver);
+        } else if(driver instanceof RecordingDriverDecorator) {
+            visit((RecordingDriverDecorator) driver);
+        } else if(driver instanceof  UsageMonitorDriverDecorator) {
+            visit((UsageMonitorDriverDecorator) driver);
+        }
+        return details;
```

Comment:
1. Visitor solution helps avoiding instanceof clauses.
2. visit methods should not be called directly (only via accept call)